index.js

```diff
@@ -26,9 +28,10 @@ var GROUP = new RegExp([
   '<([^>]*)>', // We capture the actual range expression
   ')?',
   // The trailing part is optional
-  '([^,;\\s]+)?',
+  '([^,;\\s]*)',
   ')', // end of actual group
 ].join(''), 'g');
+var HOST = /^([^\d,;\s<>]+)([\d]*)([^,;\s<>]*)$/;
```

Comment:
Removed the `[]`. I don't think we can fail on double-numbered hosts. E.g. things with custom tlds like `foo-app3.tld1` are valid hostnames we'd need to support.